web/src/main/webapp/WEB-INF/pages/application_state.jsp

```diff
@@ -110,9 +132,13 @@
                           <div class="col-sm-8">
                               <p class="form-control-static <c:if test="${stat.first}">stateData"</c:if> id="languageSkillP">${data.languageSkill}</p>
                               <c:if test="${stat.first}">
-                                  <input class="stateInput" type="text" name="languageSkill" id="languageSkillInput" style="display:none" value="${data.languageSkill}">
+                                  <input class="stateInput" type="number" name="languageSkill" id="languageSkillInput" style="display:none" value="${data.languageSkill}"
+                                  data-error="<spring:message code="candidate.error.language.incorrect"/>"
```

Comment:
to be more xml compliant, extract the <spring:message> tag before the <input> tag and assign its value to a variable (var attribute), then reference the variable like a normal variable using JSP EL.

If there are more messages like this, do the same for all of them